balancer/weightedtarget/weightedtarget_test.go

```diff
@@ -586,11 +597,13 @@ func (s) TestWeightedTarget_TwoSubBalancers_MoreBackends(t *testing.T) {
 	}
 
 	// Turn all connections down.
-	wtb.UpdateSubConnState(sc4, balancer.SubConnState{ConnectivityState: connectivity.TransientFailure})
+	wtb.UpdateSubConnState(sc4, balancer.SubConnState{
+		ConnectivityState: connectivity.TransientFailure,
+		ConnectionError:   scConnErr})
 	p = <-cc.NewPickerCh
 	for i := 0; i < 5; i++ {
-		if _, err := p.Pick(balancer.PickInfo{}); err != balancer.ErrTransientFailure {
-			t.Fatalf("want pick error %v, got %v", balancer.ErrTransientFailure, err)
+		if _, err := p.Pick(balancer.PickInfo{}); !strings.Contains(err.Error(), scConnErr.Error()) {
+			t.Fatalf("want pick error '%v', got error '%v'", scConnErr, err)
```

Comment:
You don't have to quote the formatting directive `%v`. If you want the error message to appear quoted in the output, you can use `%q` instead.